### PR TITLE
asserts: introduce the concept of sequence-forming assertion types

### PR DESCRIPTION
**Pull request by pedronis**
this is expressed via AssertionType.SequenceForming and the SequenceMember interface

so far repair and validation-set are sequence-forming

### CODE REVIEW

**Comment by pedronis** [comment]
based on #8852

File: asserts/repair.go [hunk 1]
```diff
@@ -55,6 +55,11 @@ func (r *Repair) RepairID() int {
 	return r.id
 }
 
+// Sequence implements SequenceMember, it returns the same as RepairID.
+func (r Repair) Sequence() int {
```
Comment (stolowski) [inline asserts/repair.go:59]: Is there a reason it's not pointer type (r *Repair)?

File: asserts/asserts.go [hunk 2]
```diff
@@ -65,11 +66,21 @@ func (at *AssertionType) MaxSupportedFormat() int {
 	return maxSupportedFormat[at.Name]
 }
 
+// SequencingForming returns true if the assertion type has a positive
+// intenger >= 1 as the last component (preferably called "sequence")
```
[the remark this answers is not in the document]
Comment (pedronis) [reply asserts/asserts.go:70]: thanks, mmh, is the spell checker in the static tests not working anymore??

File: asserts/repair.go (hunk 1 above)
Comment (pedronis) [reply asserts/repair.go:59]: no, it's actually wrong, thanks for spotting this

**Pull request merged**